pkg/system/phase2_creating.go

```diff
@@ -87,8 +94,17 @@ func (r *Reconciler) ReconcilePhaseCreatingForMainClusters() error {
 	if err := r.ReconcileObject(r.CoreApp, r.SetDesiredCoreApp); err != nil {
 		return err
 	}
-	if err := r.ReconcileDB(); err != nil {
-		return err
+	// create the mongo db only if mongo db url is not given.
+	if r.NooBaa.Spec.MongoDbURL == "" {
+		if err := r.UpgradeSplitDB(); err != nil {
+			return err
+		}
+		if err := r.ReconcileDB(); err != nil {
```

Comment:
ReconcileDB also reconciles postgres. you should only refer to `MongoDbURL` when dealing with mongo. I think for now you can verify in phase1 that if `MongoDbURL` is passed then the `DB_TYPE` must be mongo. later on, we might need to handle the migration from external mongo to postgres, where the db-type is postgres and there is a mongoURL